deploy/operator/gather.sh

```diff
@@ -17,9 +17,9 @@ function gather_hive_data() {
 }
 
 function gather_olm_data() {
-  oc get catalogsource "${ASSISTED_SERVICE_OPERATOR_CATALOG}" -n openshift-marketplace -o yaml > ${LOGS_DEST}/${ASSISTED_SERVICE_OPERATOR_CATALOG}.log
-  oc get subscription assisted-service-operator -n "${ASSISTED_NAMESPACE}" -o yaml > ${LOGS_DEST}/assisted-service-operator-subscription.log
-  oc get installplan -n "${ASSISTED_NAMESPACE}" -o yaml  > ${LOGS_DEST}/oc_install_plan.log
+  oc get catalogsource "${ASSISTED_SERVICE_OPERATOR_CATALOG}" -n openshift-marketplace -o yaml > ${LOGS_DEST}/${ASSISTED_SERVICE_OPERATOR_CATALOG}.log || true
```

Comment:
maybe we should apply it to the 3rd line (``set -xeo pipefail``) and just make the whole script optional in its run?
because I think there's no justification to make some lines optional and the others not

Reply:
I agree very much. I find it generally confusing when I see Prow telling me "e2e-ai-operator-ztp-disconnected-assisted-baremetal-operator-gather failed" because I'd expect -gather to be just an opportunistic step

Reply:
How about having just `set -x` for this particular script? Anything against solving the problem like that?

Reply:
yes sounds good

Reply:
on the job level it is best-effort btw: https://github.com/openshift/release/blob/eec4d78a6279dcc32cd3b622783297b59920eb59/ci-operator/step-registry/assisted/baremetal/operator/ztp/assisted-baremetal-operator-ztp-workflow.yaml#L4